.github/workflows/publish_image_chart.yaml

```diff
@@ -76,69 +102,15 @@ jobs:
           password: ${{ secrets.GCR_SA }}
 
       # Building and pushing container images.
-      - name: Build and push pipecd image
-        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
-        with:
-          push: true
-          context: .
-          file: cmd/pipecd/Dockerfile
-          platforms: linux/amd64,linux/arm64
-          tags: ${{ env.GHCR }}/pipe-cd/pipecd:${{ env.PIPECD_VERSION }}
-      - name: Build and push piped image
-        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
-        with:
-          push: true
-          context: .
-          file: cmd/piped/Dockerfile
-          platforms: linux/amd64,linux/arm64
-          tags: |
-            ${{ env.GHCR }}/pipe-cd/piped:${{ env.PIPECD_VERSION }}
-            ${{ env.GCR }}/pipecd/piped:${{ env.PIPECD_VERSION }}
-      - name: Build and push piped-okd image
-        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
-        with:
-          push: true
-          context: .
-          file: cmd/piped/Dockerfile-okd
-          platforms: linux/amd64,linux/arm64
-          tags: ${{ env.GHCR }}/pipe-cd/piped-okd:${{ env.PIPECD_VERSION }}
-      - name: Build and push launcher image
-        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+      - name: Build and push ${{ matrix.image }} image
+        uses: docker/build-push-action@0565240e2d4ab88bba5387d719585280857ece09 #v5.0.0
         with:
           push: true
           context: .
-          file: cmd/launcher/Dockerfile
+          file: ${{ matrix.dockerfile }}
           platforms: linux/amd64,linux/arm64
-          tags: |
-            ${{ env.GHCR }}/pipe-cd/launcher:${{ env.PIPECD_VERSION }}
-            ${{ env.GCR }}/pipecd/launcher:${{ env.PIPECD_VERSION }}
-      - name: Build and push launcher-okd image
-        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
-        with:
-          push: true
-          context: .
-          file: cmd/launcher/Dockerfile-okd
-          platforms: linux/amd64,linux/arm64
-          tags: ${{ env.GHCR }}/pipe-cd/launcher-okd:${{ env.PIPECD_VERSION }}
-      - name: Build and push pipectl image
-        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
-        with:
-          push: true
-          context: .
-          file: cmd/pipectl/Dockerfile
-          platforms: linux/amd64,linux/arm64
-          tags: ${{ env.GHCR }}/pipe-cd/pipectl:${{ env.PIPECD_VERSION }}
-      - name: Build and push helloworld image
-        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
-        with:
-          push: true
-          context: .
-          file: cmd/helloworld/Dockerfile
-          platforms: linux/amd64,linux/arm64
-          tags: |
-            ${{ env.GHCR }}/pipe-cd/helloworld:${{ env.PIPECD_VERSION }}
-            ${{ env.GCR }}/pipecd/helloworld:${{ env.PIPECD_VERSION }}
```

Comment:
It looks that some container images are hosted by GCR 👀 
This may be because Cloud Run can use only with Artifact Registry, GCR, Docker Hub.
So it would be nice to consider for GCR 🤔 
WDYT @khanhtc1202 ?

📝 memo
Cloud Run docs: https://cloud.google.com/run/docs/deploying
PRs that maybe related
- https://github.com/pipe-cd/pipecd/pull/3763
- https://github.com/pipe-cd/pipecd/pull/3772

Reply:
@ffjlabo good catch!
if we GCR only when Cloud Run, so ideally we could bring all images to GCR
however, not sure if PipeCD want to rely on a specific vendor?
@khanhtc1202 

btw, even we won't bring all the images to GCR, we could add more configuration for the matrix by this way 
https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#example-adding-configurations

already push the change base on that, please check below comment